pkg/bff/server.go

```diff
@@ -172,12 +174,26 @@ func (s *Server) SetHealth(health bool) {
 	log.Debug().Bool("healthy", health).Msg("server health set")
 }
 
+func (s *Server) SetURL(url string) {
+	s.Lock()
+	s.url = url
+	s.Unlock()
+	log.Debug().Str("url", url).Msg("server url set")
+}
```

Comment:
Wrapping in the lock prevents a race condition in order to get our tests to pass.